packages/elements/list/src/getListDeleteForward.ts

```diff
@@ -22,15 +30,52 @@ export const getListDeleteForward = (editor: SPEditor) => {
   const res = getListItemEntry(editor, {});
 
   let moved: boolean | undefined = false;
-  if (!isSelectionAtBlockEnd(editor) || !res) {
+  if (!isSelectionAtBlockEnd(editor)) {
     return moved;
   }
 
+  if (!res) {
+    const afterPoint = Editor.after(editor, editor.selection!.focus.path);
+
+    if (afterPoint) {
+      // there is a block after it
+      const nextSiblingListRes = getListItemEntry(editor, {
+        at: afterPoint.path,
+      });
+
+      if (nextSiblingListRes) {
+        // the next block is a list
+        const { listItem } = nextSiblingListRes;
+        const parentBlockEntity = getBlockAbove(editor, {
+          at: editor.selection!.anchor,
+        });
+
+        if (!getText(editor, parentBlockEntity![1])) {
+          // the selected block is empty
+          Transforms.removeNodes(editor, { at: editor.selection!.focus });
+
+          return true;
+        }
+
+        if (hasListChild(editor, listItem[0])) {
+          // the next block has children, so we have to move the first item up
+          const sublistRes = getListItemEntry(editor, {
+            at: [...listItem[1], 1, 0, 0],
+          });
+
+          moveListItemUp(editor, sublistRes!);
+        }
+      }
+    }
+    return false;
+  }
+
```

Comment:
All transformations should be inside `Editor.withoutNormalizing`.

Reply:
I know what it does, it's not about whether it's needed, it's more about using it uniformly. There are a log of bugs in plate still most probably due to uncalled withoutNormalizing. At some time, someone might remove withoutNormalizing from moveListItemUp, or someone might add another transformation and forget to add it. Plate in general is not calling withoutNormalizing in a strategic way, it's calls are mostly ad-hoc. Anyway, i strongly feel that all transformations should be called inside withoutNormalizing as a matter of convention, as it caused too many issues so far to be ignored.

Reply:
Agreed on that, I feel it's a bit redundant to call withoutNormalizing in all transforms. Any idea of a shorter reusable way to do it?

Reply:
Any logic which might trigger more than one transform should be wrapped in withoutNormalizing, otherwise various bugs can happen. If one was to extrapolate on that, it would be simple to state that all transforms should probably be called within withoutNormalizing, as the mental overhead of thinking whether to and possibly forgetting to do so outweighs the performance cost of calling the function, if there is a cost to calling the function to begin with. We had various problems due to this in the list plugin, it's not a theoretical issue.

It might be redundant, but the alternative is not having it on multiple transforms and causing subtle bugs. People will not remember to use it on multiple transforms (there are examples of this in plate), so might as well make it a rule :shrug: What i'm suggesting for the time being is a convention to avoid bugs, as an overall solution is well outside the scope of this PR. There can, of course, be talk about an overarching solution, i just don't think this PR is the time to do it.

On the other hand, this transform already has a withoutNormalizing call, so given all the factors i see little reason to not use it.

Reply:
I meant that I agree on using withoutNormalizing on all transforms, sorry for the confusion 🤭

That it would be nice if we find a short helper we can use in all transforms (which calls withoutNormalizing), but can't think of any right now

Reply:
@nemanja-tosic I get your point and agree, you are right an extra `withoutNormalizing` will make less of a mess down the line, if someone does not pay attention..
I will make a refactor on this and everything will be wrapped in a `withoutNormalizing`.